horreum-backend/src/main/java/io/hyperfoil/tools/horreum/changedetection/HunterEDivisiveModel.java

```diff
@@ -0,0 +1,280 @@
+                        } else {
+                            log.errorf("Could not find datapoint in set!");
+                        }
+                    } else {
+                        log.errorf("Could not parse hunter line: '%s'", changeDetails);
+                    }
+                }
+            }
+
+        } else {
+            log.debugf("No change points were detected in : %s", tmpFiles.tmpdir.getAbsolutePath());
+        }
+    }
+
+    protected boolean validateInputCsv(TmpFiles tmpFiles) {
+        executeProcess(tmpFiles, true, "bash", "-l", "-c", "hunter validate");
+
+        try(FileReader fileReader = new FileReader(tmpFiles.logFile);
+            BufferedReader reader = new BufferedReader(fileReader);){
+
+            Optional<String> optLine = reader.lines().filter(line -> line.contains("Validation finished")).findFirst();
+            if(optLine.isEmpty()) {
+                log.errorf("Could not validate: %s", tmpFiles.tmpdir.getAbsolutePath());
+                return false;
+            }
+            if( optLine.get().contains("INVALID") ) {
+                log.errorf("Invalid format for: %s; see log for details: %s", tmpFiles.tmpdir.getAbsolutePath(), tmpFiles.logFile.getAbsolutePath());
+                return false;
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return true;
+    }
+
+    @Override
+    public ModelType getType() {
+        return ModelType.BULK;
+    }
+
+
+    protected static class TmpFiles {
+        final File inputFile;
+        final File tmpdir;
+        final File confFile;
+        final File logFile;
+
+        public static TmpFiles instance() throws IOException {
+            return new TmpFiles();
+        }
+
+        public TmpFiles() throws IOException {
+            tmpdir = Files.createTempDirectory("hunter").toFile();
+
+            Path respourcesPath = Path.of(tmpdir.getAbsolutePath(), "tests", "resources");
+            Files.createDirectories(respourcesPath);
+            inputFile = Path.of(respourcesPath.toFile().getAbsolutePath(), "horreum.csv").toFile();
+
+            confFile = Path.of(respourcesPath.toFile().getAbsolutePath(), "hunter.yaml").toFile();
+            logFile = Path.of(respourcesPath.toFile().getAbsolutePath(), "hunter.log").toFile();
+
+            try (InputStream confInputStream = HunterEDivisiveModel.class.getClassLoader().getResourceAsStream("changeDetection/hunter.yaml")) {
+
+
+                try( OutputStream confOut = new FileOutputStream(confFile)){
+                    confOut.write(confInputStream.readAllBytes());
+                } catch (IOException e){
+                    log.error("Could not extract Hunter configuration from archive");
+                }
+
+            } catch (IOException e) {
+                log.error("Could not create temporary file for Hunter eDivisive algorithm", e);
+            }
```

Comment:
In both these catch clauses we are just logging errors without forwarding the exception, is this intended? because it looks like those operations should not throw any error to make the `Hunter` algorithm succeed, right? In that case I am expecting to block the process earlier raising an exception also there. Am I missing something?

Reply:
I need to think about this, we *should* handle it failing, but this is an async process that is not related to a user request. 

Right now, it would be system admins that would be investigating these issues, so writing the error to the log should be fine. For other async processes in Horreum, we log errors in `PersistentLogDAO` entities (for actions, reports, transformations etc) that are made available in the UI.

I have an issue open to improve this: https://github.com/Hyperfoil/Horreum/issues/905
For now, I think we can log any problem(s), and need to revisit how async processes outcomes are reported back to the User

Reply:
@lampajr I have added a change detection log table: https://github.com/Hyperfoil/Horreum/pull/1494/commits/ef07be7bba889c5e3ac0a3836dceabe260d996a2

There is no UI for users to view the errors, but it should help system admins identify the root cause of issues

Reply:
That is great, esp for the first implementation